gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -69,21 +85,41 @@ var gpii = fluid.registerNamespace("gpii");
                 funcName: "gpii.lifecycleManager.getSession",
                 args: ["{that}", "{arguments}.0"] // user token
             },
-            // TODO: refactor these three methods so that they return promises
-            // TODO: really do it!
+            processQueue: {
+                funcName: "gpii.lifecycleManager.processQueue",
+                args: ["{that}", "{that}.queue"]
+            },
+            addToQueue: {
+                funcName: "gpii.lifecycleManager.addToQueue",
+                args: ["{that}", "{that}.queue", "{arguments}.0"]
+            },
+            processStart: {
+                funcName: "gpii.lifecycleManager.processStart",
+                args: [ "{that}", "{arguments}.0"]
+            },
+            processStop: {
+                funcName: "gpii.lifecycleManager.processStop",
+                args: [ "{that}", "{arguments}.0"]
+            },
+            processUpdate: {
+                funcName: "gpii.lifecycleManager.processUpdate",
+                args: [ "{that}", "{arguments}.0"]
+            },
             stop: {
-                funcName: "gpii.lifecycleManager.stop",
-                args: ["{that}", "{kettle.request}", "{arguments}.0", "{arguments}.1"]
-            },                    // options, callback
+                noexpand: true,
```

Comment:
This directive is meaningless here - it is only effective within a mergePolicy. You will be getting the functions expanded as usual before being wrapped - please see suggestion in IRC yesterday